### ISSUE TITLE
Revisit Fast APIs implementations & New implementations

### ISSUE THREAD

**Issue by H4ad** [nodejs/performance#182]
Since #23, there have been many implementations in Node using fast APIs, and I want to open this issue to encourage to revisit them and probably take a look at other places where we could implement the latest changes.

Since https://github.com/nodejs/node/pull/54077, we have now support to throw errors inside fast apis, this was not possible before and we the "options.fallback" was the only way to throw errors.
- c++ ref: https://github.com/nodejs/node/blob/main/deps/v8/src/d8/d8-test.cc#L53-L72

We also had some limitations around int64 and bigints, and now I think those limitations was solved since I found some tests on v8 returning bigints:
- c++ ref: https://github.com/nodejs/node/blob/main/deps/v8/src/d8/d8-test.cc#L1245-L1297
- js ref: https://github.com/nodejs/node/blob/main/deps/v8/test/mjsunit/compiler/fast-api-calls-64-bit-integer-values.js#L182

I'm still trying to figure out all the limitations around these new changes, so any hints around these new changes are very welcome.

**Comment by anonrig** [comment]
I think we should also start adding more tests to fast api paths. I've opened a PR for timers for example: https://github.com/nodejs/node/pull/55905

**Comment by H4ad** [comment]
I agree, I will try to open PRs related to `node_process_methods`, there are some functions that I think can be converted to fast api, but I'm not sure yet, and the ones that already have fast api enabled, we can add the tests.

The only issue  I had around adding tests for fast api was the nodejs compilation taking more than 32GB of memory, the maximum memory used was around ~50GB.
